ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -38,15 +39,58 @@ def __init__(self):
 
         # TODO: Add other member variables you need below
 
-        rospy.spin()
+	self.pose = None
+	self.base_waypoints = None
+	self.waypoints_2d = None
+	self.waypoint_tree = None
+
+	self.loop()
+
+        # rospy.spin()
+    
+    def loop(self):
```

Comment:
How about naming this something more descriptive, like periodically_publish_waypoints?